File: test/test_java/ruier/DoubleBitsTest.java
// javac ruier/DoubleBitsTest.java
// java ruier.DoubleBitsTest

package ruier;



public class DoubleBitsTest {

    public static void main(String args[]) {
            int bits = 2023 << 52;
    System.out.println(Double.longBitsToDouble(bits)); // 1.048046261E-314
    }
}

File: test/test_java/ruier/DoubleBitsTest.dart

// dart ruier/DoubleBitsTest.dart

main() {
    int bits = 2023 << 52;
    print(bits); // 91 1078 2046 17051 3408

    var cc = BigInt.from(2023);
    var ccc = cc<<55; // 72886256369364107264
    print(ccc);
  }
